packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/mira/MiraController.java

```diff
@@ -81,6 +82,35 @@ void init() {
 		taskService.addResponseHandler(sbmlToPetrinetResponseHandler);
 	}
 
+	@PostMapping("/amr-to-mmt")
+	@Secured(Roles.USER)
+	@Operation(summary = "convert AMR to MIRA model template")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Dispatched successfully", content = @Content(mediaType = "application/json", schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = TaskResponse.class))),
+			@ApiResponse(responseCode = "500", description = "There was an issue dispatching the request", content = @Content)
+	})
+	public ResponseEntity<JsonNode> convertAMRtoMMT(@RequestBody final JsonNode model) {
+		try {
+			final TaskRequest req = new TaskRequest();
+			req.setType(TaskType.MIRA);
+			req.setInput(objectMapper.writeValueAsString(model).getBytes());
+			req.setScript(AMRToMMTResponseHandler.NAME);
+
+			// send the request
+			final TaskResponse resp = taskService.runTaskSync(req);
+			final JsonNode mmtInfo = objectMapper.readValue(resp.getOutput(), JsonNode.class);
+			return ResponseEntity.ok().body(mmtInfo);
+		} catch (final Exception e) {
+			final String error = "Unable to dispatch task request";
+			log.error("Unable to dispatch task request {}: {}", error, e.getMessage());
+			throw new ResponseStatusException(
+					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
+					error);
```

Comment:
Because these exceptions are now shown on the front end, is there any risk around `e.getMessage()` being too verbose?

Reply:
Maybe, can revisit if people complains.